### ISSUE TITLE
Product editing: Can't save changes to basic fields in non-core products

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#2780]
**Describe the bug**

As of 5.0 I expect to be able to edit basic fields for non-core products (added in https://github.com/woocommerce/woocommerce-ios/pull/2694). However, even though the basic fields are editable in the app, updating the product always returns an error.

Error in the app logs:

```
2020/09/07 17:13:39:236  🔵 Tracked product_detail_update_error, properties: [AnyHashable("error_description"): "Error Domain=Yosemite.ProductUpdateError Code=3 \"(null)\"", AnyHashable("is_wpcom_store"): true, AnyHashable("error_domain"): "Yosemite.ProductUpdateError", AnyHashable("error_code"): "3", AnyHashable("blog_id"): 137210602]
2020/09/07 17:13:39:240  ⛔️ Error updating Product: The operation couldn’t be completed. (Yosemite.ProductUpdateError error 3.)
```

**To Reproduce**

1. Add the WooCommerce Bookings extension to your store. (I expect I'd get the same results with a different extension and non-core product type, but I tested with Bookings.)
2. Add at least one bookable product.
3. Go to Settings > Experimental Features, and turn on the Products switch.
4. Go to the Products tab.
5. Tap on the bookable product.
6. Try editing some of the editable fields. (I tested editing multiple fields or just a single field, same results.)
7. Tap "Update."

Result: After tapping update you get a "Cannot update product" error.

**Screenshots**
<img src="https://user-images.githubusercontent.com/8658164/92406966-431eb980-f131-11ea-841d-ef296ac849e7.PNG" width="300px">

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone XS
- iOS version: 13.6.1
- WooCommerce iOS version: 5.0 TestFlight beta

**Comment by rachelmcr** [comment]
@jaclync Do you think there could be an issue with my test store in particular or is there possibly a bug with the API involved here?

Note: My store is running Woo Core version 4.5.0-rc.3.

**Comment by jaclync** [comment]
>  Do you think there could be an issue with my test store in particular or is there possibly a bug with the API involved here?
> Note: My store is running Woo Core version 4.5.0-rc.3.

@rachelmcr I was able to reproduce the update error on my test site (WC 4.4.1) for a bookable product (Booking product type). It's interesting that the error does not occur for subscription products (simple/variable subscription) with WooCommerce Subscriptions extension.

There's an open issue #2636 to include the actual error (right now it just says `The operation couldn’t be completed. (Yosemite.ProductUpdateError error 3.)` in case of unexpected errors) - I opened a PR https://github.com/woocommerce/woocommerce-ios/pull/2781 to fix this. The error is actually on sending `type` parameter to the product update endpoint (`⛔️ Error updating Product: unknown(error: Dotcom Error: [rest_invalid_param] Invalid parameter(s): type)`). I'm guessing WooCommerce Bookings extension has a validation on this parameter, while WooCommerce Subscriptions extension doesn't.

As a short-term fix for release 5.0: we can disable encoding product type for non-core products.

Long-term fix:
Ideally, we only send the **changed** properties to the product update API and that would address this issue. Currently, we send all the fields (including the product type) to the product update endpoint because we don't have a good way to exclude the same properties since [the encode function](https://github.com/woocommerce/woocommerce-ios/blob/c04c9ca5bfc16cae697f32191ad16971f0d802bb/Networking/Networking/Model/Product/Product.swift#L415-L489) doesn't take extra parameters. I'm thinking we can filter the properties in the remote layer, where it takes in the updated product and original product. I'll p2 this proposal later today.
